packages/hoppscotch-sh-admin/src/modules/i18n.ts

```diff
@@ -1,7 +1,23 @@
-import { createI18n } from 'vue-i18n';
+import { I18n, createI18n } from 'vue-i18n';
 import { HoppModule } from '.';
 import messages from '@intlify/unplugin-vue-i18n/messages';
 
+// A reference to the i18n instance
+let i18nInstance: I18n<
+  Record<string, unknown>,
+  Record<string, unknown>,
+  Record<string, unknown>,
+  string,
+  false
+> | null = null;
+
+/**
+ * Returns the i18n instance
+ */
+export function getI18n() {
+  return i18nInstance!.global.t;
+}
+
```

Comment:
In the `userManagement.ts` file, we are using `getI18n` with out checking if it is null, which can be a possibility, if this function is used before the vue app gets ready.

we could take an additional parameter `i18nFunc` for the `handleUserDeletion` which can be passed from the component using it. so we could remove this function.

although this function will allow to use the i18n from .ts files directly without interactions from a component, whoever using it has to handle the null case appropriately.

Reply:
The `i18n` module initialization happens at the package entry point. Hence, `i18Instance` is guaranteed to have the expected value.

https://github.com/hoppscotch/hoppscotch/blob/787aab650ff584f9d9cd86aea8935c74b72bc632/packages/hoppscotch-sh-admin/src/main.ts#L62

We follow a similar approach on the app side as well.

https://github.com/hoppscotch/hoppscotch/blob/787aab650ff584f9d9cd86aea8935c74b72bc632/packages/hoppscotch-common/src/modules/i18n.ts#L127-L132